File: src/main/java/com/github/nsc/de/shake/lexer/characterinput/position/PositionMarker.java
package com.github.nsc.de.shake.lexer.characterinput.position;

public interface PositionMarker {


    // ***********************************************
    // Getters

    /**
     * The index of the position
     *
     * @return Thei ndex of the position
     *
     * @author <a href="https://github.com/nsc-de">Nicolas Schmidt &lt;@nsc-de&gt;</a>
     */
    int getIndex();

    /**
     * The column of the position
     *
     * @return The column of the position
     *
     * @author <a href="https://github.com/nsc-de">Nicolas Schmidt &lt;@nsc-de&gt;</a>
     */
    int getColumn();

    /**
     * The line of the position
     *
     * @return The line of the position
     *
     * @author <a href="https://github.com/nsc-de">Nicolas Schmidt &lt;@nsc-de&gt;</a>
     */
    int getLine();
}

File: util/src/main/kotlin/com/github/nsc/de/shake/util/characterinput/position/PositionMarker.kt
package com.github.nsc.de.shake.util.characterinput.position

interface PositionMarker {
    // ***********************************************
    // Getters
    /**
     * The index of the position
     *
     * @return The index of the position
     *
     * @author [Nicolas Schmidt &lt;@nsc-de&gt;](https://github.com/nsc-de)
     */
    val index: Int

    /**
     * The column of the position
     *
     * @return The column of the position
     *
     * @author [Nicolas Schmidt &lt;@nsc-de&gt;](https://github.com/nsc-de)
     */
    val column: Int

    /**
     * The line of the position
     *
     * @return The line of the position
     *
     * @author [Nicolas Schmidt &lt;@nsc-de&gt;](https://github.com/nsc-de)
     */
    val line: Int
}